spec/idempo_spec.rb

```diff
@@ -160,71 +169,47 @@ def close
     it "is not idempotent if the HTTP verb is different" do
       post "/", "somedata", "HTTP_X_IDEMPOTENCY_KEY" => "idem"
       expect(last_response).to be_ok
-      expect(last_response.headers["X-Foo"]).to eq("bar")
+      expect(last_response.headers["x-foo"]).to eq("bar")
       first_response_body = last_response.body
 
       patch "/", "somedata", "HTTP_X_IDEMPOTENCY_KEY" => "idem"
       expect(last_response).to be_ok
       expect(last_response.body).not_to eq(first_response_body) # response should not have been reused
     end
 
-    it "is not idempotent if the request body is different" do
-      post "/", "somedata", "HTTP_X_IDEMPOTENCY_KEY" => "idem"
-      expect(last_response).to be_ok
-      expect(last_response.headers["X-Foo"]).to eq("bar")
-      first_response_body = last_response.body
-
-      post "/", "somedata2", "HTTP_X_IDEMPOTENCY_KEY" => "idem"
-      expect(last_response).to be_ok
-      expect(last_response.headers["X-Foo"]).to eq("bar")
-      expect(last_response.body).not_to eq(first_response_body) # response should not have been reused
-    end
```

Comment:
What about the removed specs?

Reply:
Some were irrelevant - this one, for example, does not work with Rack 3 because the request body is no longer rewindable - and thus can't contribute to the fingerprint. Other tests do cover the "standard happy path" well IMO.